public_html/wp-content/themes/pattern-directory/src/components/pattern-grid/index.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * WordPress dependencies
+ */
+import { Spinner } from '@wordpress/components';
+import { store as coreStore } from '@wordpress/core-data';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import PatternThumbnail from '../pattern-thumbnail';
+
+function PatternGrid() {
+	const posts = useSelect( ( select ) => {
+		// @todo This only works for logged in users, we'll need to create our own store for general use.
+		const { getEntityRecords } = select( coreStore );
+		return getEntityRecords( 'postType', 'wporg-pattern' );
+	} );
+	// `posts` will be null while the fetch happens.
+	const isLoading = ! Array.isArray( posts );
```

Comment:
One thing to mention here is that `posts` will start as an empty array then switch to `null` when the query starts which makes sense for this component, but gives an intermediate state in the view which is more obvious in slower connections. It leads to some jank on load.

Reply:
Yeah, this is mostly placeholder for a real data infrastructure, where we could have a different selector for the `isLoading` state.